x/mint/keeper/hooks_test.go

```diff
@@ -1,198 +1,395 @@
+		Staking:          sdk.NewDecWithPrec(25, 2),
+		PoolIncentives:   sdk.NewDecWithPrec(45, 2),
+		DeveloperRewards: sdk.NewDecWithPrec(25, 2),
+		CommunityPool:    sdk.NewDecWithPrec(0o5, 2),
 	}
+)
 
-	app.EpochsKeeper.BeforeEpochStart(futureCtx, params.DistrEpochIdentifier, height)
-	app.EpochsKeeper.AfterEpochEnd(futureCtx, params.DistrEpochIdentifier, height)
-
-	lastReductionPeriod = app.MintKeeper.GetLastReductionEpochNum(ctx)
-	suite.Require().Equal(lastReductionPeriod, app.MintKeeper.GetParams(ctx).ReductionPeriodInEpochs)
-
-	for ; height < lastReductionPeriod+app.MintKeeper.GetParams(ctx).ReductionPeriodInEpochs; height++ {
-		devRewardsModuleAcc := app.AccountKeeper.GetModuleAccount(ctx, types.DeveloperVestingModuleAcctName)
-		devRewardsModuleOrigin := app.BankKeeper.GetAllBalances(ctx, devRewardsModuleAcc.GetAddress())
-		feePoolOrigin := app.DistrKeeper.GetFeePool(ctx)
-
-		app.EpochsKeeper.BeforeEpochStart(futureCtx, params.DistrEpochIdentifier, height)
-		app.EpochsKeeper.AfterEpochEnd(futureCtx, params.DistrEpochIdentifier, height)
-
-		mintParams = app.MintKeeper.GetParams(ctx)
-		mintedCoin := app.MintKeeper.GetMinter(ctx).EpochProvision(mintParams)
-		expectedRewardsCoin, err := keeper.GetProportions(mintedCoin, mintParams.DistributionProportions.Staking)
-		suite.Require().NoError(err)
-		expectedRewards := sdk.NewDecCoin("stake", expectedRewardsCoin.Amount)
-
-		// check community pool balance increase
-		feePoolNew := app.DistrKeeper.GetFeePool(ctx)
-		suite.Require().Equal(feePoolOrigin.CommunityPool.Add(expectedRewards), feePoolNew.CommunityPool, height)
+// TestAfterEpochEnd tests that the after epoch end hook correctly
+// distributes the rewards depending on what epoch it is in.
+func (suite *KeeperTestSuite) TestAfterEpochEnd() {
+	var (
+		testWeightedAddresses = []types.WeightedAddress{
+			{
+				Address: testAddressOne.String(),
+				Weight:  sdk.NewDecWithPrec(233, 3),
+			},
+			{
+				Address: testAddressTwo.String(),
+				Weight:  sdk.NewDecWithPrec(5, 1),
+			},
+			{
+				Address: testAddressThree.String(),
+				Weight:  sdk.NewDecWithPrec(50, 3),
+			},
+			{
+				Address: testAddressFour.String(),
+				Weight:  sdk.NewDecWithPrec(217, 3),
+			},
+		}
+		maxArithmeticTolerance   = sdk.NewDec(5)
+		expectedSupplyWithOffset = sdk.NewDec(0)
+		expectedSupply           = sdk.NewDec(keeper.DeveloperVestingAmount)
+	)
 
-		// test that the balance decreased by the correct amount
-		devRewardsModuleAfter := app.BankKeeper.GetAllBalances(ctx, devRewardsModuleAcc.GetAddress())
-		expectedDevRewardsCoin, err := keeper.GetProportions(mintedCoin, mintParams.DistributionProportions.DeveloperRewards)
-		suite.Require().NoError(err)
-		suite.Require().Equal(devRewardsModuleAfter.Add(expectedDevRewardsCoin), devRewardsModuleOrigin, expectedRewards.String())
-	}
-}
+	suite.assertAddressWeightsAddUpToOne(testWeightedAddresses)
 
-func (suite *KeeperTestSuite) TestMintedCoinDistributionWhenDevRewardsAddressEmpty() {
```

Comment:
Note to reviewers: This test case is covered at a lower level of abstraction:
https://github.com/osmosis-labs/osmosis/blob/789835db3ea96f75720b9a7640e729bb9f89fc9d/x/mint/keeper/keeper_test.go#L597-L602

No need for it here